### ISSUE TITLE
Plotting with xlim sufficiently far from origin creates Image size too large

### ISSUE THREAD

**Issue by AmmoniumX** [sympy/sympy#26377]
TLDR: It looks like plot() is currently implemented by first generating an image from x=0 all the way to the x-axis limits, and _then_ cropping it to the x limits, while it should preferably generate the image only within the x-axis limits to avoid this error. Note xlim doesn't necessarily need to define a large boundary, it just needs to be sufficiently far away from the x=0 origin.

When creating a plot with xlim too far away from the origin (x=0), it will either throw a `ValueError` on a Jupyter notebook or an UserWarning with an empty plot if run from interpreter.

```python
from sympy import symbols
from sympy.plotting import plot

x = symbols('x')
center = 1_000_000

p = plot(x, xlim=(center - 1, center + 1), show=False)
p.show()
```
Notice how our x-axis limits (center - 1, center + 1) are pretty small, of only length=2.

If we run this code as-is through `python example.py`, we get `UserWarning: Tight layout not applied. The left and right margins cannot be made large enough to accommodate all axes decorations.`, with this generated figure:
![Figure_1](https://github.com/sympy/sympy/assets/75765000/a2ca099f-86db-4a13-a7e9-4c7cc9a6c27e)

If we run from IPython/Jupyter notebook, no figure is generated and we get **ValueError**: Image size of 248000357x389 pixels is too large. It must be less than 2^16 in each direction.:
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
File [c:\Users\diego\AppData\Local\Programs\Python\Python310\lib\site-packages\IPython\core\formatters.py:338](file:///C:/Users/diego/AppData/Local/Programs/Python/Python310/lib/site-packages/IPython/core/formatters.py:338), in BaseFormatter.__call__(self, obj)
    [336](file:///C:/Users/diego/AppData/Local/Programs/Python/Python310/lib/site-packages/IPython/core/formatters.py:336)     pass
    [337](file:///C:/Users/diego/AppData/Local/Programs/Python/Python310/lib/site-packages/IPython/core/formatters.py:337) else:
--> [338](file:///C:/Users/diego/AppData/Local/Programs/Python/Python310/lib/site-packages/IPython/core/formatters.py:338)     return printer(obj)
    [339](file:///C:/Users/diego/AppData/Local/Programs/Python/Python310/lib/site-packages/IPython/core/formatters.py:339) # Finally look for special method names
    [340](file:///C:/Users/diego/AppData/Local/Programs/Python/Python310/lib/site-packages/IPython/core/formatters.py:340) method = get_real_method(obj, self.print_method)

File [c:\Users\diego\AppData\Local\Programs\Python\Python310\lib\site-packages\IPython\core\pylabtools.py:152](file:///C:/Users/diego/AppData/Local/Programs/Python/Python310/lib/site-packages/IPython/core/pylabtools.py:152), in print_figure(fig, fmt, bbox_inches, base64, **kwargs)
    [149](file:///C:/Users/diego/AppData/Local/Programs/Python/Python310/lib/site-packages/IPython/core/pylabtools.py:149)     from matplotlib.backend_bases import FigureCanvasBase
    [150](file:///C:/Users/diego/AppData/Local/Programs/Python/Python310/lib/site-packages/IPython/core/pylabtools.py:150)     FigureCanvasBase(fig)
--> [152](file:///C:/Users/diego/AppData/Local/Programs/Python/Python310/lib/site-packages/IPython/core/pylabtools.py:152) fig.canvas.print_figure(bytes_io, **kw)
    [153](file:///C:/Users/diego/AppData/Local/Programs/Python/Python310/lib/site-packages/IPython/core/pylabtools.py:153) data = bytes_io.getvalue()
    [154](file:///C:/Users/diego/AppData/Local/Programs/Python/Python310/lib/site-packages/IPython/core/pylabtools.py:154) if fmt == 'svg':

File [c:\Users\diego\AppData\Local\Programs\Python\Python310\lib\site-packages\matplotlib\backend_bases.py:2338](file:///C:/Users/diego/AppData/Local/Programs/Python/Python310/lib/site-packages/matplotlib/backend_bases.py:2338), in FigureCanvasBase.print_figure(self, filename, dpi, facecolor, edgecolor, orientation, format, bbox_inches, pad_inches, bbox_extra_artists, backend, **kwargs)
   [2334](file:///C:/Users/diego/AppData/Local/Programs/Python/Python310/lib/site-packages/matplotlib/backend_bases.py:2334) try:
   [2335](file:///C:/Users/diego/AppData/Local/Programs/Python/Python310/lib/site-packages/matplotlib/backend_bases.py:2335)     # _get_renderer may change the figure dpi (as vector formats
   [2336](file:///C:/Users/diego/AppData/Local/Programs/Python/Python310/lib/site-packages/matplotlib/backend_bases.py:2336)     # force the figure dpi to 72), so we need to set it again here.
   [2337](file:///C:/Users/diego/AppData/Local/Programs/Python/Python310/lib/site-packages/matplotlib/backend_bases.py:2337)     with cbook._setattr_cm(self.figure, dpi=dpi):
-> [2338](file:///C:/Users/diego/AppData/Local/Programs/Python/Python310/lib/site-packages/matplotlib/backend_bases.py:2338)         result = print_method(
   [2339](file:///C:/Users/diego/AppData/Local/Programs/Python/Python310/lib/site-packages/matplotlib/backend_bases.py:2339)             filename,
   [2340](file:///C:/Users/diego/AppData/Local/Programs/Python/Python310/lib/site-packages/matplotlib/backend_bases.py:2340)             facecolor=facecolor,
   [2341](file:///C:/Users/diego/AppData/Local/Programs/Python/Python310/lib/site-packages/matplotlib/backend_bases.py:2341)             edgecolor=edgecolor,
...
---> [87](file:///C:/Users/diego/AppData/Local/Programs/Python/Python310/lib/site-packages/matplotlib/backends/backend_agg.py:87) self._renderer = _RendererAgg(int(width), int(height), dpi)
     [88](file:///C:/Users/diego/AppData/Local/Programs/Python/Python310/lib/site-packages/matplotlib/backends/backend_agg.py:88) self._filter_renderers = []
     [90](file:///C:/Users/diego/AppData/Local/Programs/Python/Python310/lib/site-packages/matplotlib/backends/backend_agg.py:90) self._update_methods()

ValueError: Image size of 248000357x389 pixels is too large. It must be less than 2^16 in each direction.
```

A working version using matplotlib directly is given here, but it would be nice if sympy's plot function would work under these circumstances:
```python
from sympy import symbols
import numpy as np
import matplotlib.pyplot as plt

x = symbols('x')
center = 1_000_000

X = np.linspace(center - 1, center + 1, 400)
Y = [x.subs(x, val).evalf() for val in X]

# Plotting
plt.figure(figsize=(10, 6))
plt.plot(X, Y)
plt.show()
```

**Comment by shishir-11** [comment]
Maybe the following code snippet can solve your problem 

```python
from sympy import symbols
from sympy.plotting import plot

x = symbols('x')
center = 1_000_000

p = plot(x, (x,center - 1, center + 1), show=False)
p.show()
```
I think the issue is happening because the default range defined is from (-10,10) and while using xlim we define the x axis position and not the range. 
To get the result we can also use 
```python
 p = plot(x,(x,center-10,center+10),xlim=(center-10,center+10),show=False)
```

Maybe a check can be written allowing the range to be changed according to region overlap with the limits on the x-axis.

**Comment by sylee957** [comment]
I think that you are using wrong option `xlim` to control the range of the plots. I'd explain that `xlim` just crops out the range of already drawn plot somewhere, and it should work something like image cropping.

**Comment by AmmoniumX** [comment]
> Maybe the following code snippet can solve your problem
> 
> ```python
> from sympy import symbols
> from sympy.plotting import plot
> 
> x = symbols('x')
> center = 1_000_000
> 
> p = plot(x, (x,center - 1, center + 1), show=False)
> p.show()
> ```
> 
> I think the issue is happening because the default range defined is from (-10,10) and while using xlim we define the x axis position and not the range. To get the result we can also use
> 
> ```python
>  p = plot(x,(x,center-10,center+10),xlim=(center-10,center+10),show=False)
> ```
> 
> Maybe a check can be written allowing the range to be changed according to region overlap with the limits on the x-axis.

This method does indeed work, but I feel that if a user is calling `plot(x, xlim=(a, b))`, the intention is clearly to construct a plot image around the specified x-axis limits. I don't see why the current implementation of "make a big image in the backend, then crop it to the user's specified limits" would be desirable. The function could interpret the given xlim param as a range when no range is given.

**Comment by sylee957** [comment]
I assume that you are suggesting to do something like
```
if range is None and xlim is not None:
    range = xlim
return plot(...)
```
However, we are already doing many complicated things to `xlim`

https://github.com/sympy/sympy/blob/031fb0196ce344380c488b8d6a178ed167ff890c/sympy/plotting/backends/matplotlibbackend/matplotlib.py#L221

and I'm afraid that adding more branches there, just increases the complexity of implementation, than giving much significant benefits.
